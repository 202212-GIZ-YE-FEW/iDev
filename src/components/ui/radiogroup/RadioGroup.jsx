import React from "react";
import { Children } from "react";

export default function RadioGroup(props) {
    const { title, children, asButton = false } = props;
    React.Children.forEach(children, (child) => {
        if (child.type.name !== "RadioInputItem") {
            throw new Error(
                "RadioGroup component can only have RadioInputItem as children"
            );
        }
    });
    return (
        <div className='p-4'>
            {title && (
                <p className='mb-4 p-2 text-3xl font-normal break-words text-black'>
                    {title}
                </p>
            )}
            <fieldset className='space-y-4'>
                <div className='text-9xl font-normal text-black'>
                    {Children.map(children, (child) => {
                        return React.cloneElement(child, { asButton });
                    })}
                </div>
            </fieldset>
        </div>
    );
}
