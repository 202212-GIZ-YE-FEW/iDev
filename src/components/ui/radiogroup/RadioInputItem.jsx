import clsx from "clsx";

export default function RadioInputItem(props) {
    const { id, name, value, checked = false, onChange, content, as } = props;

    const inputProps = {
        id,
        name,
        value,
        checked,
        onChange,
    };

    const labelClassNames = clsx({
        "w-full py-2 ms-2 text-base lg:text-xl font-medium text-black ps-3":
            as === "standard",
        "w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white":
            as === "button",
        "block w-full h-full cursor-pointer peer-checked:border-[5px] peer-checked:border-cyan peer-checked:border-dashed peer-checked:rounded-lg":
            as === "card",
    });

    return (
        <div
            className={clsx("", {
                "flex items-center": as === "standard",
                "flex items-center space-y-4": as === "button",
                "h-full rounded-md min-h-[14rem]": as === "card",
            })}
        >
            <input
                type='radio'
                {...inputProps}
                className={clsx({
                    "w-5 h-5 bg-black border-black": as === "standard",
                    "peer hidden": as === "button" || as === "card",
                })}
            />
            {content && (
                <label htmlFor={id} className={labelClassNames}>
                    {content}
                </label>
            )}
        </div>
    );
}
