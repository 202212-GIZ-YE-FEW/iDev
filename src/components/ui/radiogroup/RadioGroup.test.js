import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import RadioGroup from "./RadioGroup";
import RadioInputItem from "./RadioInputItem";

it("renders correctly", () => {
    const tree = renderer.create(<RadioGroup />).toJSON();
    expect(tree).toMatchSnapshot();
});
it("renders title correctly", () => {
    const { getByText } = render(<RadioGroup title='test title' />);
    expect(getByText("test title")).toBeInTheDocument();
});
// it("renders children correctly", () => {
//     const { getByText } = render(
//         <RadioGroup title='test title'>
//             <RadioInputItem title='test title for child' />
//         </RadioGroup>
//     );
//     expect(getByText("test title for child")).toBeInTheDocument();
// });
// it("passes asButton prop to child", () => {
//     const { getByText } = render(
//         <RadioGroup title='test title'>
//             <RadioInputItem title='test title for child' />
//         </RadioGroup>
//     );
//     expect(getByText("test title for child")).toBeInTheDocument();
// });
it("does not render title if not passed", () => {
    const { queryByText } = render(
        <RadioGroup>
            <RadioInputItem title='test title for child' />
        </RadioGroup>
    );
    expect(queryByText("test title")).not.toBeInTheDocument();
});
