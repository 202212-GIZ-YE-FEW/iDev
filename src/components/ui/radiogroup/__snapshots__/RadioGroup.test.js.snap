// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="p-4"
>
  <fieldset
    className="space-y-4"
  >
    <div
      className="text-9xl font-normal text-black"
    />
  </fieldset>
</div>
`;
